File: Generacidad/Ejercicio-1/Caja.java

class Caja<T> {
    private T contenido;

    public void guardar(T objeto) {
        this.contenido = objeto;
    }

    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("Hola mundo");

        Caja<Integer> cajaNumero = new Caja<>();
        cajaNumero.guardar(123);

        System.out.println("Contenido de la caja de texto: " + cajaTexto.obtener());
        System.out.println("Contenido de la caja de número: " + cajaNumero.obtener());
    }
}

File: Generacidad/Ejercicio-1/Caja.py
from typing import Generic, TypeVar

T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self):
        self.contenido: T = None

    def guardar(self, objeto: T):
        self.contenido = objeto

    def obtener(self) -> T:
        return self.contenido

if __name__ == "__main__":
    caja_texto = Caja[str]()
    caja_texto.guardar("Hola mundo")

    caja_numero = Caja[int]()
    caja_numero.guardar(123)

    print("Contenido de la caja de texto:", caja_texto.obtener())
    print("Contenido de la caja de número:", caja_numero.obtener())